jdaviz/configs/specviz/plugins/parsers.py

```diff
@@ -246,3 +269,30 @@ def combine_lists_to_1d_spectrum(wl, fnu, dfnu, wave_units, flux_units):
     spec = Spectrum1D(flux=fnuall * flux_units, spectral_axis=wlall * wave_units,
                       uncertainty=unc)
     return spec
+
+
+def split_spectrum_with_2D_flux_array(data):
+    """
+    Helper function to split Spectrum1D of 2D flux to a SpectrumList of nD objects.
+    Parameters
+    ----------
+    data : `~specutils.Spectrum1D`
+        Spectrum with 2D flux array
+
+    Returns
+    -------
+    new_data : `~specutils.SpectrumList`
+        list of unpacked spectra
+    """
+    new_data = []
+    for i in range(data.flux.shape[0]):
+        unc = None
+        mask = None
+        if data.uncertainty is not None:
+            unc = data.uncertainty[i, :]
+        if data.mask is not None:
+            mask = data.mask[i, :]
+        new_data.append(Spectrum1D(flux=data.flux[i, :], spectral_axis=data.spectral_axis,
```

Comment:
What about the other properties like redshift, metadata, etc? Do they not have to be propagated?

Reply:
Also... is an utility function like this better suited for specutils upstream?

Reply:
I just added the meta, I had discussed with Brian and we agreed that it's ok to attach the meta from the original to each split spectrum at least for now. Redshift comes along with the `spectral_axis`. 

I think I'd prefer to keep this here for now, but we could potentially upstream it at some point. I don't know exactly where I'd want to put it in `specutils`.

Reply:
OK, I opened https://github.com/astropy/specutils/issues/1188